File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> v) {
    int answer = 0;
    stack<int> s;
    for(int i=0;i<v.size();i++){
        if(s.empty()) s.push(v[i]==1);
        else if(!s.empty()){
            if((s.top()==1 && v[i]==2)||(s.top()==2 && v[i]==3)) s.push(v[i]);
            else if(s.top()==3 && v[i]==1){
                answer++;
                s.pop(); s.pop(); s.pop();
            }
            else s.push(v[i]==1);
        }
    }

    return answer;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<Integer> s = new Stack<>();
        for(int x : ingredient){
            if(s.empty()) s.push((x==1?1:0));
            else{
                if((s.peek()==1&&x==2)||(s.peek()==2&&x==3)) s.push(x);
                else if(s.peek()==3&&x==1){
                    answer++;
                    s.pop(); s.pop(); s.pop();
                }
                else{
                    s.push((x==1?1:0));
                }
            }
        }
        return answer;
    }
}